### ISSUE TITLE
Comments should be allowed above hierarchical selectors in indentation

### ISSUE THREAD

**Issue by magsout** [stylelint/stylelint#940]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

it's a bug

> Which rule, if any, is this issue related to?

`indentation`

> What CSS is needed to reproduce this issue?

e.g.

```css
:root {
  --r-Grid-baseFontSize: 1rem;
  --r-Grid-gutter: 1rem;
}

.r-Grid {
  padding: 0;
  margin: 0;
}

  .r-Grid-cell {
    font-size: var(--r-Grid-baseFontSize);
    width: 100%;
    padding: 0;
  }

  /* Modifier --withGutter */
  .r-Grid--withGutter {
    margin: 0 calc(-0.5 * var(--r-Grid-gutter));
  }

    .r-Grid--withGutter > .r-Grid-cell {
      padding: 0 calc(0.5 * var(--r-Grid-gutter));
    }
```

> What stylelint configuration is needed to reproduce this issue?

```json
"stylelint": {
    "extends": "stylelint-config-standard",
    "rules": {
      "indentation": [ 2, {
        "hierarchicalSelectors": true
        }]
    }
  },
```

> Which version of stylelint are you using?

`stylelint@5.0.1`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

`stylelint ./lib/base.css`

> What did you expect to happen?

 "No warnings to be flagged."

> What actually happened (e.g. what warnings or errors you are getting)?

```shell
lib/base.css
17:3	Expected indentation of 0 spaces (indentation)
18:3	Expected indentation of 0 spaces (indentation)
19:5	Expected indentation of 2 spaces (indentation)
20:3	Expected indentation of 0 spaces (indentation)
```

If I removed my comment `/* Modifier --withGutter */` there is no more warning.

**Comment by davidtheclark** [comment]
Open to a PR to fix this for hierarchical selectors.

**Comment by davidtheclark** [comment]
We've decided to deprecate this feature of indentation because it is a lot of added complexity for a very specific and not-too-widespread practice (https://github.com/stylelint/stylelint/issues/1309). Not saying it's a *bad* practice, just that it's probably not a good use of the core team's time to maintain this feature.

If you would like to create a plugin that enforces hierarchical indentation, that would be awesome and let us know about it!
